File: 20th Sep Task/20thSepTask_EmployeeAPISoln/EmployeeDomainModel/Models/Employee.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EmployeeDomainModel.Models
{
    public class Employee
    {
        public Employee()
        {
            Skills = new HashSet<Skill>();
        }
        public int Employee_Id { get; set; }
        public string Name { get; set; }
        public string Status { get; set; }
        public string Manager { get; set; }
        public string WFM_Manager { get; set; }
        public string Email { get; set; }
        public string Lockstatus { get; set; }
        public decimal Experience { get; set; }
        public virtual ICollection<Skill> Skills { get; set; }

    }
}

File: AspDotNetTraining_FinalTask/WFM_Web_AppSolution/WFM_Web_App/wwwroot/js/Employee.js
$.get("https://localhost:7115/api/Employee", function (data, status) {

    let code = "";
    for (let x in data) {
        code += "<tr>"
        code += "<td>" + data[x].employeeID + "</td>"
        code += "<td>" + data[x].name + "</td>"
        code += "<td>" + data[x].status + "</td>"
        code += "<td>" + data[x].manager + "</td>"
        code += "<td>" + data[x].wfm_Manager + "</td>"
        code += "<td>" + data[x].email + "</td>"
        code += "<td>" + data[x].experience + "</td>"
        code += "<td>"
        for (let s in data[x].skills)
            code += data[x].skills[s] + "  "
        code += "</td>"
        code += "<td> <a href= SoftLockRequestData?employee_ID class=btn btn-primary> RequestLock </a> </td> </tr>"
    }
    $('#tbody').html(code)
})

